### ISSUE TITLE
Statefulset discovery

### ISSUE THREAD

**Issue by mattniedelman** [solo-io/service-mesh-hub#791]
Hi,

I noticed pods generated by statefulsets error out while being scanned by the mesh-discovery process 
```
{"level":"warn","ts":1592596339.3222463,"logger":"mesh-discovery","msg":"Error scanning pod ice-kafka-zookeeper-0.ice-kafka: ReplicaSet.apps \"ice-kafka-zookeeper\" not found","version":"0.5.0","compute-target":"dc2"}
```
I think it might be due to the owner_fetcher only looking for replicasets and deployments:
https://github.com/solo-io/service-mesh-hub/blob/master/pkg/mesh-discovery/discovery/mesh-workload/k8s/owner_fetcher.go

The documentation here suggests that it shouldn't matter what kind of object generates the pods: https://docs.solo.io/service-mesh-hub/latest/reference/api/mesh_workload/#discovery.smh.solo.io.MeshWorkloadSpec.KubeController , so maybe it's worth just arbitrarily climbing the owner references?  

As a temporary workaround, it looks like it's be possible to generate the meshworkload custom resources manually, but I haven't tested this beyond seeing that the corresponding serviceentries are created in remote clusters

**Comment by harveyxia** [comment]
Hi Matt,

We're currently working on an overhaul of the discovery system which will take this feature into account (as well handling workloads backed by DaemonSets). It's difficult to give a confident ETA on when that will be released. If you need this feature sooner, let me know and I can implement something temporary for the short term.

**Comment by mattniedelman** [comment]
Awesome, I'll keep an eye out.  No need for a temp solution, thanks though.

**Comment by joekelley** [comment]
Hey @mattniedelman, the overhaul @harveyxia mentioned above has been released in v0.7.1. Please try it out and reopen this issue if you run into this problem again.
